src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -63,6 +63,16 @@ public SyntaxTree SyntaxTree
         protected abstract SyntaxTree SyntaxTreeCore { get; }
 
         /// <summary>
+        /// Returns true if this is a SemanticModel that ignores accessibility rules when answering semantic questions.
+        /// </summary>
+        public bool IsSuppressingAccessChecks { get; internal set; }
+
+        /// <summary>
+        /// Creates a SemanticModel that ignores accessibility rules when answering semantic questions.
+        /// </summary>
+        public abstract SemanticModel WithSuppressAccessChecks();
```

Comment:
I'd much rather expose this as a new optional parameter added to the existing Compilation.GetSemanticModel overloads which takes a Boolean flag for whether accessibility checks should be suppressed for the purpose of lookup which defaults to false. Perhaps bool ignoreAccessibility or suppressAccessChecks.